File: IrlCPC_2019/1.cpp
#include<bits/stdc++.h>
#define pb push_back
#define fi first
#define se second
#define X(x) real(x)
#define Y(x) imag(x)
using namespace std;
typedef long long ll;
typedef vector<int> vi;
typedef complex<double> cc;
typedef vector<cc> poly;

double cross(cc& a, cc& b){
    return X(a) * Y(b) - Y(a) * X(b);
}

double area(poly& p){
    double sm = 0.0;
    for(int i = 0; i < p.size(); ++i){
	sm += cross(p[i], p[(i+1)%p.size()]);
    }
    return abs(sm)/2.0;
}

int main(){
    int N; cin >> N;
    double x, y;
    poly p;
    for(int i = 0; i < N; ++i){
	cin >> x >> y;
	x *= 1000.0;
	y *= 1000.0;
	p.pb({x,y});
    }
    p.pb(p[0]);
    cout << fixed << setprecision(10) << area(p)/1e6 << endl;
    return 0;
}

File: IrlCPC_2019/1.py
def cross(a, b):
    return a[0]* b[1] - a[1] * b[0];

def area(p):
    sm = 0
    for i in range(len(p)):
        sm += cross(p[i], p[(i+1)%len(p)])
    return abs(sm)

def parse(s):
    s = ''.join(s.split('.'))
    while len(s) < 16:
        s += '0'
    return int(s)

N = int(input())
p = []
for _ in range(N):
    x, y = map(parse, input().split())
    p.append((x,y))
p.append(p[0])

b = area(p)

A = 10**30

ans = 0
for x in range(1001):
    if 2*x*A <= b*1000:
        ans = x/1000.0
    else:
        break
print(ans)
